lib/spack/spack/solver/asp.py

```diff
@@ -1969,8 +1969,9 @@ class Body:
         # TODO: use better semantics.
         arch = spec.architecture
         if arch:
-            if arch.platform:
-                clauses.append(f.node_platform(spec.name, arch.platform))
+            if arch._platforms:
+                for platform in arch._platforms:
+                    clauses.append(f.node_platform(spec.name, platform))
```

Comment:
`spack spec x platform=linux,cray` currently still fails, but I also don't really wanna complicate that.

Reply:
This part shouldn't be hard to model -- can you add it?  The platform semantics in `concretize.lp` are only 4 rules at the moment... if you need help working this in either @alalazo or I could help.